### PR TITLE
Staging folder

### PR DESCRIPTION
**Pull request by steveoh**
## Description of Changes

add `stagingDestination` to the config.

add `staging_rack` to Pallet with value from `stagingDestination` to allow for different staging destination locations without having to modify pallet/crate paths.

### Test results and coverage

```
Name                 Stmts   Miss     Cover   Missing
-----------------------------------------------------
forklift\arcgis.py      65      3    95.38%   48, 96-98
forklift\cli.py        224     64    71.43%   138, 173-178, 185, 232-235, 283-300, 307-318, 328-377
forklift\core.py       244     11    95.49%   84-88, 155, 196, 204, 286, 295, 363, 438
forklift\lift.py       131     26    80.15%   45-46, 134-155, 161, 179-180, 216-218, 231-232
forklift\models.py     170      6    96.47%   79, 262, 270, 322, 357-358
-----------------------------------------------------
TOTAL                  925    110    88.11%

5 files skipped due to complete coverage.
-----------------------------------------------------------------------------
151 tests run in 114.008 seconds.
11 skipped (140 tests passed)
```

### Speed test results

```
Speed Test Results
Dry Run: 2.35 minutes
Repeat: 70.27 seconds
```

### CODE REVIEW

**Comment by steveoh** [comment]
This would help solve #147 but doesn't really go the full mile. you'll still have to inspect the configuration property and modify it based on the value.

File: readme.md [hunk 1]
```diff
@@ -1,51 +1,65 @@
...
+## Usage
+
+The work that forklift does is defined by [Pallets](src/forklift/models.py). `forklift.models.Pallet` is a base class that allows the user to define a job for forklift to perform by creating a new class that inherits from `Pallet`. Each pallet should have `Pallet` in it's file name and be unique from it's other pallets.
+
+A Pallet can have zero or more [Crates](src/forklift/models.py). A `forklift.models.Crate` is a class that defines data that needs to be moved from one location to another (reprojecting to web mercator by default). Crates are created by calling the `add_crates` (or `add_crate`) methods within the `build` method on the pallet. For example:
```
Comment (stdavis) [inline readme.md]: "~~A~~ `forklift.models.Crate` is a class..."

File: readme.md [hunk 2]
```diff
@@ -1,51 +1,65 @@
...
 ```
 If the property is a list then the value is appended to the existing list.
 
-### Install to First Successful Run
+## Install to First Successful Run
+
 1. `pip install .\` from the directory containing `setup.py`.
 1. `forklift config init`
 1. `forklift config set --key copyDestinations --value c:\\MapData` - This is where you want your output placed.
+1. `forklift config set --key stagingDestinations --value c:\\staging` - This is where your data is staged for copy.
```
Comment (stdavis) [inline readme.md]: This is optional, right? There is a default value. I would think that we should remove it to provide the shortest path to a successful first run.

Comment (steveoh) [inline readme.md]: this is equally as optional as warehouse

**Comment by stdavis** [comment]
> This would help solve #147 but doesn't really go the full mile. you'll still have to inspect the configuration property and modify it based on the value.

Actually, I think that this solves it entirely since Staging and Production have separate config files. Or am I missing something?

File: readme.md (hunk 2 above)
Comment (stdavis) [inline readme.md]: Setting the warehouse path isn't part of these steps, is it?

Comment (steveoh) [inline readme.md]: oh i see

**Pull request merged**